ipvs.py

```diff
@@ -552,18 +577,45 @@ def get_pools(self):
         for msg in self.nlsock.query(req):
             svc_lst = msg.get_attr_list().get('service')
             service = Service.from_attr_list(svc_lst)
-            dests = []
-            out_msg = IpvsMessage(
-                'get_dest', flags=netlink.MessageFlags.MATCH_ROOT_REQUEST,
-                attr_list=IpvsCmdAttrList(service=svc_lst)
-            )
-            for dst_msg in self.nlsock.query(out_msg):
-                dst_lst = dst_msg.get_attr_list().get('dest')
-                dests.append(Dest.from_attr_list(dst_lst, svc_lst.get('af')))
-
+            dests = self.get_dests(svc_lst)
             pools.append(Pool.from_args(
                 service=service,
                 dests=dests
             ))
 
         return pools
+
+    def get_pool(self, svc_lst):
+        s = self.get_service(svc_lst)
+        if s is None:
+            return None
+        dests = self.get_dests(s.to_attr_list())
+        return Pool.from_args(service=s, dests=dests)
+
+    def get_service(self, svc_lst):
+        out_msg = IpvsMessage(
+            'get_service', flags=netlink.MessageFlags.REQUEST,
+            attr_list=IpvsCmdAttrList(service=svc_lst))
+        try:
+            res = self.nlsock.query(out_msg)
+            svc_lst = res[0].get_attr_list().get('service')
+            return Service.from_attr_list(svc_lst)
+        except RuntimeError:
+            # If the query failed because the service is not present
+            # simply return None
+            return None
+
+    def get_dests(self, svc_lst):
+        assert isinstance(svc_lst, IpvsServiceAttrList)
+        dests = []
+        out_msg = IpvsMessage(
+            'get_dest', flags=netlink.MessageFlags.MATCH_ROOT_REQUEST,
+            attr_list=IpvsCmdAttrList(service=svc_lst)
+        )
+        try:
+            for dst_msg in self.nlsock.query(out_msg):
+                dst_lst = dst_msg.get_attr_list().get('dest')
+                dests.append(Dest.from_attr_list(dst_lst, svc_lst.get('af')))
+            return dests
+        except RuntimeError:
```

Comment:
I would expect this to only happen when there is no service right?

I dont  think we should return an empty list, this would mean that the service exists but has no destinations attached to it.
I think we would be better of either bubbling up the exception or logging and returning None.

For now, I would be happy with returning None.
For logging, we obviously have been terrible at it and we should start thinking about adding some :)